implementations/rust/ockam/ockam_command/tests/bats/nodes_unit.bats

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+# ===== SETUP
+
+setup() {
+  load load/base.bash
+  load_bats_ext
+  setup_home_dir
+}
+
+teardown() {
+  teardown_home_dir
+}
+
+# ===== TESTS
+
+@test "node - can recreate a background node after it was stopped" {
+  n="$(random_str)"
+  run "$OCKAM" node create "$n"
+  assert_success
+
+  # Fail to create a node with the same name
+  run "$OCKAM" node create "$n"
+  assert_failure
+
+  run "$OCKAM" node stop "$n"
+  assert_success
+
+  # Recreate node
+  run "$OCKAM" node create "$n"
+  assert_success
+}
+
+@test "node - can recreate a foreground node after it was stopped" {
+  n="$(random_str)"
+  $OCKAM node create $n -f &
+  sleep 0.1
+  run "$OCKAM" node show "$n"
+  assert_success
+
+  # Fail to create a node with the same name
+  run "$OCKAM" node create "$n" -f
+  assert_failure
+
+  run "$OCKAM" node stop "$n"
```

Comment:
good to add these tests cases :+1:  

for a future improvement,  would be good to kill -9  the running node instead/in addition of calling  `node stop`.  That will rule out the possibility of recreate working because of some cleanup done by a "clean" stop of the node instead of a machine restart, etc.

Reply:
So, you mean manually killing the process and then try to recreate it right? If that process doesn't exist anymore then it should proceed as usual and recreate the node. I think that code is already there, I'll try to add the test!

Reply:
yes exactly but I'd say totally fine to not include that on this PR

Reply:
I'll leave it for another PR. I have the tests in place but it looks like there is a problem on foreground nodes and they don't handle the situation correctly.